src/core/storage/native-scrobbler-notification.ts

```diff
@@ -0,0 +1,42 @@
+import * as BrowserStorage from '@/core/storage/browser-storage';
+
+export default class NativeScrobblerNotification {
+	private storage = BrowserStorage.getStorage(
+		BrowserStorage.NATIVE_SCROBBLER_NOTIFICATION,
+	);
+
+	/**
+	 * Add connector ID to list of connectors that has been notified about.
+	 *
+	 * @param id - ID of connector to add
+	 */
+	public async saveHasNotified(id: string): Promise<void> {
+		if (!id) {
+			return;
+		}
+		let data = await this.storage.getLocking();
+		if (!data) {
+			data = {};
+		}
+		data[id] = true;
+		await this.storage.setLocking(data);
+	}
+
+	/**
+	 * @param id - ID of connector to check
+	 *
+	 * @returns true if notification has not been given for connector; false otherwise
+	 */
+	public async shouldNotifyAboutNativeScrobbler(
+		id: string | undefined | null,
```

Comment:
When can this be either `undefined` or `null`? isn't `id` manditory for connectors? Should we create a type to enforce this?

Reply:
Removed this, probably triggered this some other way before and moved the function call somewhere id isnt null later.